psydac/linalg/tests/test_block.py

```diff
@@ -52,23 +53,140 @@ def test_block_linear_operator_serial_dot( n1, n2, p1, p2, P1, P2  ):
     cart = CartDecomposition(D, npts, global_starts, global_ends, pads=[p1,p2], shifts=[1,1])
 
     # Create vector spaces, stencil matrices, and stencil vectors
-    V = StencilVectorSpace( cart )
-    M1 = StencilMatrix( V, V )
+    V = StencilVectorSpace( cart, dtype=dtype )
+    M1 = StencilMatrix( V, V)
     M2 = StencilMatrix( V, V )
     M3 = StencilMatrix( V, V )
     x1 = StencilVector( V )
     x2 = StencilVector( V )
 
     # Fill in stencil matrices based on diagonal index
+    if dtype==complex:
+        f=lambda k1,k2: 10j*k1+k2
+    else:
+        f=lambda k1,k2: 10*k1+k2
+
     for k1 in range(-p1,p1+1):
         for k2 in range(-p2,p2+1):
-            M1[:,:,k1,k2] = 10*k1+k2
-            M2[:,:,k1,k2] = 10*k1+k2+2.
-            M3[:,:,k1,k2] = 10*k1+k2+5.
+            M1[:,:,k1,k2] = f(k1,k2)
+            M2[:,:,k1,k2] = f(k1,k2)+2.
+            M3[:,:,k1,k2] = f(k1,k2)+5.
+
     M1.remove_spurious_entries()
     M2.remove_spurious_entries()
     M3.remove_spurious_entries()
 
+    # Fill in vector with random values, then update ghost regions
+    for i1 in range(n1):
+        for i2 in range(n2):
+            x1[i1,i2] = 2.0*random() - 1.0
+            x2[i1,i2] = 5.0*random() - 1.0
+    x1.update_ghost_regions()
+    x2.update_ghost_regions()
+
+    W = BlockVectorSpace(V, V)
+    assert W.dimension == 2*n1*n2
+    assert W.dtype == dtype
+    assert W.spaces == (V,V)
+    assert W.parallel == False
+    assert W.starts == [(0,0),(0,0)]
+    assert W.ends == [(n1-1,n2-1),(n1-1,n2-1)]
+    assert W.pads == (p1,p2)
+    assert W.n_blocks == 2
+    assert W.connectivity== {}
+#===============================================================================
+@pytest.mark.parametrize( 'dtype', [float, complex] )
+@pytest.mark.parametrize( 'n1', [8, 16] )
+@pytest.mark.parametrize( 'n2', [8, 12] )
+@pytest.mark.parametrize( 'p1', [1, 2, 3] )
+@pytest.mark.parametrize( 'p2', [1, 2, 3] )
+@pytest.mark.parametrize( 'P1', [True, False] )
+@pytest.mark.parametrize( 'P2', [True, False] )
+
+def test_2D_block_vector_serial_init( dtype, n1, n2, p1, p2, P1, P2  ):
+    # set seed for reproducibility
+    seed(n1*n2*p1*p2)
+
+    D = DomainDecomposition([n1,n2], periods=[P1,P2])
+
+    # Partition the points
+    npts = [n1,n2]
+    global_starts, global_ends = compute_global_starts_ends(D, npts)
+
+    cart = CartDecomposition(D, npts, global_starts, global_ends, pads=[p1,p2], shifts=[1,1])
+
+    # Create vector spaces, stencil matrices, and stencil vectors
+    V = StencilVectorSpace( cart, dtype=dtype )
+    x1 = StencilVector( V )
+    x2 = StencilVector( V )
+
+    # Fill in vector with random values, then update ghost regions
+    for i1 in range(n1):
+        for i2 in range(n2):
+            x1[i1,i2] = 2.0*random() - 1.0
+            x2[i1,i2] = 5.0*random() - 1.0
+    x1.update_ghost_regions()
+    x2.update_ghost_regions()
```

Comment:
If you are not checking the contents of `x1` and `x2`, why are you filling these vectors?

Reply:
[86b06e9](https://github.com/pyccel/psydac/pull/273/commits/86b06e97fda7a028f2d6ec3f23b6c81d44eaf696)